EarlGrey/Core/GREYElementInteraction.m

```diff
@@ -232,13 +232,6 @@ - (instancetype)performAction:(id<GREYAction>)action error:(__strong NSError **)
       [defaultNotificationCenter postNotificationName:kGREYDidPerformActionNotification
                                                object:nil
                                              userInfo:actionUserInfo];
-      // If we encounter a failure and going to raise an exception, raise it right away before
```

Comment:
but we want to raise exception right away before any part of the UI is altered. Failures should capture and report the exact state at which things failed.